src/main/java/com/google/devtools/build/lib/query2/aquery/ActionGraphTextOutputFormatterCallback.java

```diff
@@ -330,4 +333,34 @@ private Map<String, String> getParamFileNameToContentMap() {
     }
     return paramFileNameToContentMap;
   }
+
+  /**
+   * Decode a bytestring that might contain UTF-8, and escape any characters
+   * outside the basic printable ASCII range.
+   *
+   * This function is intended for human consumption in debug output that needs
+   * to be durable against unusual encoding settings, and does not guarantee
+   * that the escaping process is reverseable.
+   *
+   * Non-printable ASCII characters are formatted as `{U+00XX}`. Characters
```

Comment:
Let's go with the standard-ish `\uXXXX` notation to make the output copy-pastable into a script.

Reply:
Done.